src/BSON/Value.stub.php

```diff
@@ -0,0 +1,210 @@
+     */
+    public const TYPE_DECIMAL128 = 0x13;
+
+    /**
+     * @var int
+     * @cvalue BSON_TYPE_MAXKEY
+     */
+    public const TYPE_MAXKEY = 0x7F;
+
+    /**
+     * @var int
+     * @cvalue BSON_TYPE_MINKEY
+     */
+    public const TYPE_MINKEY = 0xFF;
+
+    private function __construct() {}
+
+    final public function getType(): int {}
+
+#if PHP_VERSION_ID >= 80000
+    final public function getValue(): mixed {}
+#else
+    /** @return mixed */
+    final public function getValue() {}
+#endif
+
+#if PHP_VERSION_ID >= 80000
+    final static public function fromPHP(mixed $value): Value {}
+#else
+    /** @param mixed $value */
+    final static public function fromPHP($value): Value {}
+#endif
+
+    /** Accessors */
+    final public function getArray(): PackedArray {}
+    final public function getBinary(): Binary {}
+    final public function getBool(): bool {}
+    final public function getCode(): Javascript {}
+    final public function getDateTime(): UTCDateTime {}
+    final public function getDBPointer(): DBPointer {}
+    final public function getDecimal128(): Decimal128 {}
+    final public function getDocument(): Document {}
+    final public function getDouble(): float {}
+    final public function getInt32(): int {}
+    final public function getInt64(): Int64 {}
+    final public function getMaxKey(): MaxKey {}
+    final public function getMinKey(): MinKey {}
+    final public function getNull(): null {}
+    final public function getObjectId(): ObjectId {}
+    final public function getRegex(): Regex {}
+    final public function getSymbol(): Symbol {}
+    final public function getTimestamp(): Timestamp {}
+    final public function getUndefined(): Undefined {}
+    final public function getUtf8(): string {}
+
+    /** Type checkers */
+    final public function isArray(): bool {}
+    final public function isBinary(): bool {}
+    final public function isBool(): bool {}
+    final public function isCode(): bool {}
+    final public function isDateTime(): bool {}
+    final public function isDBPointer(): bool {}
+    final public function isDecimal128(): bool {}
+    final public function isDocument(): bool {}
+    final public function isDouble(): bool {}
+    final public function isInt32(): bool {}
+    final public function isInt64(): bool {}
+    final public function isMaxKey(): bool {}
+    final public function isMinKey(): bool {}
+    final public function isNull(): bool {}
+    final public function isObjectId(): bool {}
+    final public function isRegex(): bool {}
+    final public function isSymbol(): bool {}
+    final public function isTimestamp(): bool {}
+    final public function isUndefined(): bool {}
+    final public function isUtf8(): bool {}
+
+    final public static function __set_state(array $properties): Value {}
+
+    final public function serialize(): string {}
+
+    /** @param string $serialized */
+    final public function unserialize($serialized): void {}
```

Comment:
I realize this is consistent with every other stub, but why do we have an `@param` annotation here?

The interface has a `string` type since https://github.com/php/php-src/commit/273731fb762d661fb40015afd851ab340cf7a429#diff-c8adc7ea754be4705d22956277447f2a8cdcd19e3185caa1513391d864a82384R44

It had no type hint in 7.4: https://github.com/php/php-src/blob/dca8d5565b947270a29f232bc54efd9df3b92b94/Zend/zend_interfaces.c#L572-L580

Given that, should we be using `#if PHP_VERSION_ID >= 80000` to conditionally declare a type hint? If so, can you create a ticket for that?

Reply:
This is due to the stub tool (gen_stub.php), which complains about parameters without types. Each parameter either needs to have a declared type or a corresponding `@param` annotation. The same goes for return types, where a return type or `@return` annotation are necessary.